subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/text/TablePrinter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics.internal.text;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import org.gradle.internal.logging.text.StyledTextOutput;
+
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.gradle.internal.logging.text.StyledTextOutput.Style.Header;
+import static org.gradle.internal.logging.text.StyledTextOutput.Style.Normal;
+
+/**
+ * Simple utility for printing a table.
+ */
+public class TablePrinter {
```

Comment:
I wonder if this should be changed to extend `org.gradle.reporting.ReportRenderer`, and have, in addition to an inner `Row` class, an inner `TableModel` class which would be a value type containing the fields in here, such that the `render` method could be given the model and the `StyledTextOutput`.  This class seems to be combining 2 responsibilities.  It's hardly a big deal, since this isn't meant for reuse.  Just a thought as I was reading this.